db/commitlog/commitlog.cc

```diff
@@ -1898,6 +2011,10 @@ future<> db::commitlog::segment_manager::shutdown() {
         } catch (...) {
             p = std::current_exception();
         }
+
+        send(shutdown_request{});
+        co_await std::move(_message_loop);
```

Comment:
What prevents other messages to be posted after this one?

Reply:
The shutdown logic has at this point eliminated all code paths that can generate messages. In the changes further on, we of course move this ensuring inside the loop itself, so "shutdown" message is then not necessarily the last message, it just changes state and starts trying to dismantle everything.